network/src/dms.rs

```diff
@@ -331,25 +316,119 @@ impl<N: GossipNetwork, S: Storage> DistributedMessageSet<N, S> {
+    pub async fn broadcast_all(
+        &self,
+        network_config: &NetworkConfig,
+        known_peers: &[Peer],
+    ) -> Result<(), Error> {
+        let mut tasks1 = Vec::new();
+        let state = self.read_state().await?;
+        let messages = self
+            .read_messages()
+            .await?
+            .into_iter()
+            .map(RawMessage::from_message)
+            .collect::<Vec<_>>();
+        let height = state.height;
+
+        for peer in known_peers {
+            let port_key = state.key.clone();
+            let messages_ = messages.clone();
+            let task = async move {
+                let stub = DistributedMessageSetRpcInterfaceStub::new(Box::new(HttpClient::new(
+                    format!(
+                        "{}:{}/dms",
+                        peer.address.ip(),
+                        peer.ports
+                            .get(&port_key)
+                            .ok_or_else(|| anyhow!("can't find port key: {}", port_key))?
+                    ),
+                    reqwest::Client::new(),
+                )));
+                stub.add_messages(height, messages_.clone())
+                    .await?
+                    .map_err(|e| anyhow!(e))?;
+                Result::<(), Error>::Ok(())
+            };
+            tasks1.push((task, format!("RPC message add to {}", peer.public_key)));
+        }
+        let tasks2 = messages.into_iter().map(|message| {
+            let network_config = network_config.clone();
+            let peers = known_peers.to_owned();
+            let message_hash = message.data.to_hash256();
+            (
+                async move {
+                    N::broadcast(
+                        &network_config,
+                        &peers,
+                        serde_json::to_vec(&message).unwrap(),
+                    )
+                    .await?;
+                    Result::<(), Error>::Ok(())
+                },
+                format!("broadcast message {} to all peers", message_hash),
+            )
+        });
+        let mut tasks = Vec::new();
+        let mut messages = Vec::new();
+        for (task, msg) in tasks1 {
+            tasks.push(task.boxed());
+            messages.push(msg);
+        }
+        for (task, msg) in tasks2 {
+            tasks.push(task.boxed());
+            messages.push(msg);
+        }
+        let results = future::join_all(tasks).await;
+        for (result, msg) in results.into_iter().zip(messages.iter()) {
+            if let Err(e) = result {
+                log::warn!("failure in {}: {}", msg, e);
+            }
+        }
+        Ok(())
+    }
+
     /// Reads the messages from the storage.
     pub async fn read_messages(&self) -> Result<Vec<Message>, Error> {
-        let result = read_messages(&*self.storage.read().await).await?;
-        Ok(result)
+        let files = self.storage.read().await.list_files().await?;
```

Comment:
Why don't we filter `STATE_FILE_PATH ` here?

Reply:
fixed.